pkg/job_controller/job.go

```diff
@@ -251,7 +251,17 @@ func (jc *JobController) ReconcileJobs(
 		return result, err
 	}
 
-	jc.Metrics.AllPodsLaunchDelaySeconds(pods, metaObject, jobStatus)
+	// Metering first pod launch delay when job state transit from created to running.
+	if commonutil.IsCreated(*oldStatus) && commonutil.IsRunning(jobStatus) {
+		jc.Metrics.FirstPodLaunchDelaySeconds(activePods, metaObject, jobStatus)
+	}
+
+	// Metring all pods launch delay when latest pods are all active after reconciled, and previous
+	// job status has not reached a all-active state.
+	if (k8sutil.GetTotalAvtiveReplicas(jobStatus.ReplicaStatuses) == totalReplicas) &&
```

Comment:
If the there are pods failed and re-created while the job is running.
these two conditions can also be true I think .

Reply:
done, with extra  condition `!IsRestarting(oldStatus)`.